lib/src/modules/gates.dart

```diff
@@ -772,13 +772,20 @@ class ReplicationOp extends Module
 
   /// Constructs a ReplicationOp
   ///
-  /// The bit [bit] will be repeated over the [_width] as an output.
+  /// The signal [original] will be repeated over the [_multiplier] times as an
+  /// output.
+  /// Input [_multiplier] cannot be negative or zero, an exception will be
+  /// thrown, otherwise.
   /// [Module] is in-lined as SystemVerilog, it will use {width{bit}}
-  ReplicationOp(Logic bit, this._width)
-      : _inputName = 'input_${bit.name}',
-        _outputName = Module.unpreferredName('output_${bit.name}') {
-    addInput(_inputName, bit);
-    addOutput(_outputName, width: _width);
+  ReplicationOp(Logic original, this._multiplier)
+      : _inputName = 'input_${original.name}',
+        _outputName = Module.unpreferredName('output_${original.name}') {
+    if (_multiplier < 1) {
+      throw Exception('Multiplier({_multiplier}) is not >= 1');
```

Comment:
We're trying to make exceptions more specific (see the exceptions/ directory).  Try to avoid using the base `Exception` and instead either use an existing one or make a new one.